samples/set-spatial-reference/src/main/java/com/esri/arcgismaps/sample/setspatialreference/components/SetSpatialReferenceViewModel.kt

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.setspatialreference.components
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.Basemap
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.layers.ArcGISMapImageLayer
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel for the "Set Spatial Reference" sample.
+ */
+class SetSpatialReferenceViewModel(application: Application) : AndroidViewModel(application) {
+
+    // A simple message dialog view model for presenting errors
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Create a map using World Bonne spatial reference (WKID: 54024)
+    var arcGISMap = ArcGISMap(spatialReference = SpatialReference(wkid = WORLD_BONNE_WKID))
+
+    init {
+        viewModelScope.launch {
+                // Use an ArcGISMapImageLayer as the basemap's base layer so the
+                // map image service will be displayed in the map's spatial reference.
+                arcGISMap.setBasemap(Basemap(baseLayer = ArcGISMapImageLayer(url = WORLD_CITIES_MAP_SERVICE)))
```

Comment:
This could be placed outside of the viewModelScope and applied to the `arcGISMap`: 

```suggestion
```